Don't decompose copy (sic)

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89464
* __->__ #89461

This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Differential Revision: [D41460988](https://our.internmc.facebook.com/intern/diff/D41460988)

ezyang committed:
Don't decompose copy (sic)

This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Don't decompose copy (sic)"


This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Don't decompose copy (sic)"


This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bdhirsh commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1535,6 +1535,8 @@
 - func: copy(Tensor self, Tensor src, bool non_blocking=False) -> Tensor
   variants: function
+  dispatch:
+    CompositeExplicitAutogradNonFunctional: copy
```
I remember a while back we were debating whether or not to make `copy()` a primitive, and we settled on "we can represent copy() with existing operators, so we shouldn't bother adding a new primitive unnecessarily".

I'm mostly just wondering with that in the background - what's the idea behind making it a primitive now? Now that we care about accurate striding, representing the decomp accurately is sufficiently more annoying that it would be easier for inductor to lower it directly?

ezyang replied:
It is very annoying to write a decomposition that is both (1) functional and (2) preserves striding exactly.

bdhirsh replied:
Can definitely believe that :)

bdhirsh reviewed (approved):
I buy that

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460988).

ezyang committed:
Update on "Don't decompose copy (sic)"


This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D41460988](https://our.internmc.facebook.com/intern/diff/D41460988)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460988).

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460988).

ezyang committed:
Update on "Don't decompose copy (sic)"


This might cause problems for inductor, I want to see what they
are.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D41460988](https://our.internmc.facebook.com/intern/diff/D41460988)

[ghstack-poisoned]
